wgpu-types/src/lib.rs

```diff
@@ -4248,7 +4271,10 @@ pub struct TextureDescriptor<L> {
     pub format: TextureFormat,
     /// Allowed usages of the texture. If used in other ways, the operation will panic.
     pub usage: TextureUsages,
-    // TODO: missing view_formats https://www.w3.org/TR/webgpu/#dom-gputexturedescriptor-viewformats
+    /// Specifies what view format values will be allowed when calling create_view() on this texture.
+    /// Note: Adding a format to this list may have a significant performance impact,
+    /// so it is best to avoid adding formats unnecessarily.
+    pub view_formats: Option<Vec<TextureFormat>>,
```

Comment:
I don't think the `Option` is actually necessary here.

Reply:
That means users need to use `vec![]` to set empty value, would this be inconsistent with the style of using other nullable values in wgpu?

Reply:
It should be - actually the issue is that we want it to be `&'a [TextureFormat]` so people can do `&[]` for the empty value.

Reply:
Setting the view_formats type in `wgt::TextureDescriptor` to `&'a [TextureFormat]` results in an error like this:
```
the trait bound `&[TextureFormat]: Deserialize<'_>` is not satisfied  
the following other types implement trait `Deserialize<'de>`:  
&'a [u8]  
[T; 0]  
[T; 10]  
[T; 11]  
[T; 12]  
[T; 13]  
[T; 14]  
[T; 15]  
and 26 othersrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message?3#file%3A%2F%2F%2FUsers%2Flijinlei%2FRust%2Fforks%2Fwgpu%2Fwgpu-types%2Fsrc%2Flib.rs)
```

The current implementation seems a bit complicated, not sure if it's appropriate?

Reply:
I think it's just because TextureFormat doesn't impl deserialize/serialize

Reply:
TextureFormat was already implemented deserialize/serialize.

Setting to `&'a [TextureFormat]`  also causes another problem, having to add a lifetime parameter for `Texture` causes `pub type TextureId = Id<crate::resource::Texture<Dummy>>`  also have a lifetime parameter added, and `TextureId` should not have a lifetime parameter.

My current implementation is to redefine a `TextureDescriptor` in wgpu just like  `TextureViewDescriptor`.
but in `wgpu::util::device::create_texture_with_data` need to call some `wgt::TextureDescriptor` 's functions, Causes the code to be a bit weird.